### PR TITLE
os/Log: Print colored components when colors are enabled

### PR DESCRIPTION
**Pull request by drdanz**
This isn't actually a bugfix, but it is a tiny enhancement that helps **a lot** while debugging, therefore, unless someone is against this, I'd like to merge it in the stable branch.

image

(With `YARP_COMPACT_OUTPUT`):
image

### CODE REVIEW

**Comment by Nicogene** [comment]
`YARP_COMPACT_OUTPUT` by default is `false`?

**Comment by drdanz** [comment]
> `YARP_COMPACT_OUTPUT` by default is `false`?

Yes, as it was before...

File: src/libYARP_os/src/yarp/os/Log.cpp [hunk 1]
```diff
@@ -281,6 +281,51 @@ inline const char* logTypeToBgColor(yarp::os::Log::LogType t)
...
+    // LogCallback has only the name of the component, we cannot store the color
+    // in the LogComponent without breaking the API, therefore this is the less
+    // invasive option.
+    // Anyway, this is enabled only when YARP_COLORED_OUTPUT is set, therefore
+    // it can be easily disabled if this slows down the execution.
+    static std::hash<std::string> hsh; // FIXME C++17: Use string_view
+    std::size_t comp_hash = hsh(comp_name) % 12;
+    switch (comp_hash) {
+    case 0:
+        return RED;
```
Comment (Nicogene) [inline src/libYARP_os/src/yarp/os/Log.cpp:301]: I know that it is not part of this PR but isn't better to use `constexpr char[]` instead of defines?

**Review by Nicogene — COMMENTED** [review]
I have not a strong opinion on this, personally I dislike dark font on a dark background, and this may happen for some component, maybe we can use only bright colors?

**Comment by drdanz** [comment]
> I have not a strong opinion on this, personally I dislike dark font on a dark background, and this may happen for some component, maybe we can use only bright colors?

At the moment it is using all the colors available on your terminal, but if you don't like them, you can usually modify your theme from your terminal settings... I don't think that reducing the number of colors to 7 for this reason is worth...
Also you usually have to read the debug line, not the component name, the color just helps following the execution of your program...
